core/shared/src/main/scala/zio/ZPool.scala

```diff
@@ -215,7 +215,7 @@ object ZPool {
           attempted.forEach(a => invalidated.update(_ - a)) *>
             attempted.finalizer *>
             state.modify { case State(size, free) =>
-              if (size <= range.start)
+              if (size <= range.start || free < 0)
```

Comment:
How can `free` be less than zero? Isn't that a bug that should be fixed directly?

Reply:
I believe negative values represent the number of requests awaiting in queue for an item become available.